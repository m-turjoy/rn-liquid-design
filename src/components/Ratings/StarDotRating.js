import React from 'react';
import { bool, string, number, func } from 'prop-types';
import Icon from '../MerckIcons';
import { IconTouchableWrapper, IconWrapper } from './styled';
import HalfIcon from './HalfIcon';
import colors from '../../config/colors';

const StarDotRating = ({
  activeIconColor,
  inactiveIconColor,
  iconSize,
  dot,
  onIconPress,
  active,
  disabled,
  halfIcon,
}) => (
  <IconTouchableWrapper
    onPress={onIconPress}
    disabled={disabled}
    activeOpacity={1}
  >
    <IconWrapper marginRight={6} opacity={disabled ? 0.5 : 1}>
      {halfIcon ? (
        <HalfIcon type={dot ? 'dot' : 'star'} color={activeIconColor} />
      ) : (
        <Icon
          name={dot ? 'dot' : 'star'}
          size={iconSize}
          color={active ? activeIconColor : inactiveIconColor}
        />
      )}
    </IconWrapper>
  </IconTouchableWrapper>
);

StarDotRating.propTypes = {
  activeIconColor: string,
  inactiveIconColor: string,
  iconSize: number,
  dot: bool,
  onIconPress: func,
  active: bool,
  disabled: bool,
  halfIcon: bool,
};

StarDotRating.defaultProps = {
  inactiveIconColor: colors.sensitiveGreyDarkest,
  iconSize: 24,
  dot: false,
  disabled: false,
  onIconPress: () => {},
  active: false,
  halfIcon: false,
};

export default StarDotRating;
