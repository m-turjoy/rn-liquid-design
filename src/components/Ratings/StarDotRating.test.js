import React from 'react';
import { shallow, configure } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import StarDotRating from './StarDotRating';

configure({ adapter: new Adapter() });
describe('StarDotComponent component', () => {
  describe('Renders correctly', () => {
    test('it renders Default star icon', () => {
      const tree = shallow(<StarDotRating />);
      expect(tree).toMatchSnapshot();
    });
    test('it renders Default dot icon', () => {
      const tree = shallow(<StarDotRating dot />);
      expect(tree).toMatchSnapshot();
    });
  });
});
