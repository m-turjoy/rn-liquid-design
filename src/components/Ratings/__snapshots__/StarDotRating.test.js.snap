// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`StarDotComponent component Renders correctly it renders Default dot icon 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <StarDotRating
    active={false}
    disabled={false}
    dot={true}
    halfIcon={false}
    iconSize={24}
    inactiveIconColor="#e9e9ed"
    onIconPress={[Function]}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "activeOpacity": 1,
      "children": <Styled(View)
        marginRight={6}
        opacity={1}
      >
        <Icon
          color="#e9e9ed"
          name="dot"
          size={24}
          themeName="vibrantCyan"
        />
      </Styled(View)>,
      "disabled": false,
      "onPress": [Function],
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": <Icon
          color="#e9e9ed"
          name="dot"
          size={24}
          themeName="vibrantCyan"
        />,
        "marginRight": 6,
        "opacity": 1,
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "color": "#e9e9ed",
          "name": "dot",
          "size": 24,
          "themeName": "vibrantCyan",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      "type": [Function],
    },
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "activeOpacity": 1,
        "children": <Styled(View)
          marginRight={6}
          opacity={1}
        >
          <Icon
            color="#e9e9ed"
            name="dot"
            size={24}
            themeName="vibrantCyan"
          />
        </Styled(View)>,
        "disabled": false,
        "onPress": [Function],
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": <Icon
            color="#e9e9ed"
            name="dot"
            size={24}
            themeName="vibrantCyan"
          />,
          "marginRight": 6,
          "opacity": 1,
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "color": "#e9e9ed",
            "name": "dot",
            "size": 24,
            "themeName": "vibrantCyan",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        "type": [Function],
      },
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
    "attachTo": undefined,
    "hydrateIn": undefined,
  },
}
`;

exports[`StarDotComponent component Renders correctly it renders Default star icon 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <StarDotRating
    active={false}
    disabled={false}
    dot={false}
    halfIcon={false}
    iconSize={24}
    inactiveIconColor="#e9e9ed"
    onIconPress={[Function]}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "activeOpacity": 1,
      "children": <Styled(View)
        marginRight={6}
        opacity={1}
      >
        <Icon
          color="#e9e9ed"
          name="star"
          size={24}
          themeName="vibrantCyan"
        />
      </Styled(View)>,
      "disabled": false,
      "onPress": [Function],
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": <Icon
          color="#e9e9ed"
          name="star"
          size={24}
          themeName="vibrantCyan"
        />,
        "marginRight": 6,
        "opacity": 1,
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "color": "#e9e9ed",
          "name": "star",
          "size": 24,
          "themeName": "vibrantCyan",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      "type": [Function],
    },
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "activeOpacity": 1,
        "children": <Styled(View)
          marginRight={6}
          opacity={1}
        >
          <Icon
            color="#e9e9ed"
            name="star"
            size={24}
            themeName="vibrantCyan"
          />
        </Styled(View)>,
        "disabled": false,
        "onPress": [Function],
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": <Icon
            color="#e9e9ed"
            name="star"
            size={24}
            themeName="vibrantCyan"
          />,
          "marginRight": 6,
          "opacity": 1,
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "color": "#e9e9ed",
            "name": "star",
            "size": 24,
            "themeName": "vibrantCyan",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        "type": [Function],
      },
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
    "attachTo": undefined,
    "hydrateIn": undefined,
  },
}
`;
